dotCMS/src/main/java/com/dotcms/publishing/job/SiteSearchJobImpl.java

```diff
@@ -129,182 +134,213 @@ public void run(final JobExecutionContext jobContext) throws JobExecutionExcepti
-            audit.setPagesCount(pagesCount);
-            audit.setFilesCount(filesCount);
-            audit.setUrlmapsCount(urlmapCount);
-            audit.setAllHosts(indexAll);
-            audit.setFireDate(jobContext.getFireTime());
-            audit.setHostList(UtilMethods.join(indexHosts,",",true));
-            audit.setIncremental(incremental);
-            audit.setStartDate(startDate);
-            audit.setEndDate(endDate);
-            audit.setIndexName( UtilMethods.isSet(newIndexName) ? newIndexName :  indexName );
-            audit.setJobId(jobId);
-            audit.setJobName(dataMap.getString(QUARTZ_JOB_NAME));
-            audit.setLangList(UtilMethods.join(languageToIndex,","));
-            audit.setPath(paths.size() > 0 ? UtilMethods.join(paths,",") : "/*");
-            audit.setPathInclude(include);
-            siteSearchAuditAPI.save(audit);
-        }
-        catch(DotDataException ex) {
-            Logger.error(this, "can't save audit data",ex);
-        }
-        finally {
+                final SiteSearchAudit audit = new SiteSearchAudit();
+                audit.setPagesCount(pagesCount);
+                audit.setFilesCount(filesCount);
+                audit.setUrlmapsCount(urlmapCount);
+                audit.setAllHosts(preparedJobContext.isIndexAll());
+                audit.setFireDate(jobContext.getFireTime());
+                audit.setHostList(preparedJobContext.getJoinedHosts());
+                audit.setIncremental(preparedJobContext.isIncremental());
+                audit.setStartDate(preparedJobContext.getStartDate());
+                audit.setEndDate(preparedJobContext.getEndDate());
+                audit.setIndexName(
+                        UtilMethods.isSet(preparedJobContext.getNewIndexName()) ? preparedJobContext
+                                .getNewIndexName() : preparedJobContext.getIndexName());
+                audit.setJobId(preparedJobContext.getJobId());
+                audit.setJobName(preparedJobContext.getJobName());
+                audit.setLangList(preparedJobContext.getLangList());
+                audit.setPath(preparedJobContext.getPaths());
+                audit.setPathInclude(preparedJobContext.isPathInclude());
+                siteSearchAuditAPI.save(audit);
+
+
+        } catch (DotDataException ex) {
+            Logger.error(this, "can't save audit data", ex);
+        } finally {
             HibernateUtil.closeSession();
         }
-
         date = DateUtil.getCurrentDate();
         ActivityLogger.logInfo(getClass(), "Job Finished", "User: " +userAPI.getSystemUser().getUserId()+ "; Date: " + date + "; Job Identifier: " + SiteSearchAPI.ES_SITE_SEARCH_NAME  );
         AdminLogger.log(getClass(), "Job Finished", "User: " +userAPI.getSystemUser().getUserId()+ "; Date: " + date + "; Job Identifier: " + SiteSearchAPI.ES_SITE_SEARCH_NAME );
     }
 
+     private synchronized PreparedJobContext prepareJob(final JobExecutionContext jobContext)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use block level rather than method level synchronization](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5188880)